src/plugins/discover/public/application/apps/main/discover_main_route.tsx

```diff
@@ -75,8 +75,6 @@ export function DiscoverMainRoute({ services, history }: DiscoverMainProps) {
 
     async function loadSavedSearch() {
       try {
-        // force a refresh if a given saved search without id was saved
-        setSavedSearch(undefined);
```

Comment:
By resetting `savedSeach` here, URL is not getting replaced with fresh query when navigating to a `new` from saved search.